docs/reference/function-configuration/code-entry-types.md

```diff
@@ -0,0 +1,245 @@
+
+The downloaded function code can optionally contain a **function.yaml** file with function configuration for enriching the original configuration (in the configuration file that sets the code-entry type) according to the following merge strategy:
+
+- Field values that are set only in the downloaded configuration are added to the original configuration.
+- List and map field values &mdash; such as `meta.labels` and `spec.env` &mdash;are merged by adding any values that are set only in the downloaded configuration to the values that are set in the original configuration.
+- In case of a conflict &mdash; i.e., if the original and downloaded configurations set different values for the same element &mdash; the original configuration takes precedence and the value in the downloaded configuration is ignored.
+
+<a id="code-entry-type-github"></a>
+### GitHub code-entry type (`github`)
+
+Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `github` (dashboard: **Code entry type** = `GitHub`) to download the function code from a GitHub repository. The following configuration fields provide additional information for performing the download:
+
+- `spec.build` &mdash;
+  - `path` (dashboard: **URL**) (Required) &mdash; the URL of the GitHub repository that contains the function code.
+  - `codeEntryAttributes` &mdash;
+    -  `branch` (dashboard: **Branch**) (Required) &mdash; the GitHub repository branch from which to download the function code.
+    - `headers.Authorization` (dashboard: **Token**) (Optional) &mdash; a GitHub access token for download authentication.
+    - `workDir` (dashboard: **Work directory**) (Optional) &mdash; the relative path to the function-code directory within the configured repository branch.
+
+<a id="code-entry-type-github-example"></a>
+#### Example
+
+```yaml
+spec:
+  description: my Go function
+  handler: main:Handler
+  runtime: golang
+  build:
+    codeEntryType: "github"
+    path: "https://github.com/my-organization/my-repository"
+    codeEntryAttributes:
+      branch: "my-branch"
+      headers:
+        Authorization: "my-Github-access-token"
+      workDir: "/go/myfunc"
+```
+
+<a id="code-entry-type-archive"></a>
+### Archive-file code-entry type (`archive`)
+
+Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `archive` (dashboard: **Code entry type** = `Archive`) to download a ZIP archive file of the function code either from an [Iguazio Data Science Platform](https://www.iguazio.com) ("platform") data container or from any URL that doesn't require download authentication. The following configuration fields provide additional information for performing the download:
```

Comment:
```suggestion
Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `archive` (dashboard: **Code entry type** = `Archive`) to download a ZIP archive file of the function code either from one of 2 sources:
- [Iguazio Data Science Platform](https://www.iguazio.com) ("platform") data container - authenticated
- Some other URL that doesn't require download authentication. The following configuration fields provide additional information for performing the download:
```

Maybe my suggestion is a bit janky, but I'm trying to separate the "unauthenticated" case from out native data-container url, so it is not implied that an Iguazio container use-case is unauthenticated :)

Reply:
I edited the doc in the `archive` code-entry type section, per your suggestion (but phrased a bit differently) + I also added "(authenticated)" after the platform option reference in the `archive` bullet at the start of the parent "External function-code entry types" section; (I decided to keep the description in the "Overview" section as-is). See commit 362260c.

NOTE: I think the source of the problem is the grouping into one code-entry type of the platform ("V3IO") archive file (which has an authenticated download) and all archive files that don't require authentication. I understand that originally we mean for `archive` to be used for all archive download types, including AWS S3, but when we decided to create a separate code-entry type for S3, I think we should have also created a separate type for platform downloads (or create single authenticated-download type for S3 and platform downloads and document which configuration fields are relevant for which download type.) The current unintuitive entry-types division makes it challenging to keep the doc clear yet short and not cumbersome.

Reply:
I understand, we'll make a note of it, maybe v3io should be a separate code entry type indeed, shouldn't be too much work